examples/car_fsm.erl

```diff
@@ -0,0 +1,543 @@
+         burnt = B} = S,
+  {Travelled, Acceleration, Burnt} =
+    acceleration_calculations({Speed, -Value}, Fuel),
+  S#state{fuel = Fuel - Burnt,
+          speed = Speed + Acceleration,
+          distance = Distance + Travelled,
+          burnt = B + Burnt};
+next_state_data(_, _, S, _, {call, _, travel, [Value]}) ->
+  #state{fuel = Fuel,
+         speed = Speed,
+         distance = Distance,
+         burnt = B} = S,
+  {Travelled, Burnt} = travel_calculations(Value, Speed, Fuel),
+  S#state{fuel = Fuel - Burnt,
+          distance = Distance + Travelled,
+          burnt = B + Burnt};
+next_state_data(_, _, S, _, {call, _, refuel, [Value]}) ->
+  #state{fuel = Fuel,
+         speed = Speed,
+         distance = Distance,
+         burnt = B} = S,
+  {Travelled, Acceleration, Burnt} =
+    acceleration_calculations({Speed, -Speed}, Fuel),
+  S#state{fuel = Fuel - Burnt + Value,
+          speed = Speed + Acceleration,
+          distance = Distance + Travelled,
+          burnt = B + Burnt}.
+
+postcondition(stopped, stopped, _, {call, _, refuel, _}, {0.0, 0.0}) ->
+  true;
+postcondition(_, _, _, _, {Distance, Burnt}) ->
+  Distance >= 0.0 andalso Burnt >= 0.0.
+
+weight(_, _, {call, _, brake, _}) -> 2;
+weight(_, _, {call, _, travel, _}) -> 4;
+weight(_, _, _) -> 1.
+
+
+%% -----------------------------------------------------------------------------
+%% proper_fsm helpers
+%% -----------------------------------------------------------------------------
+
+
+accelerate_commands(Speed) ->
+  [{slow, ?CALL(accelerate, [accelerator(Speed, slow)])}
+   || Speed < ?SLOW_THRESHOLD] ++
+    [{normal, ?CALL(accelerate, [accelerator(Speed, normal)])}
+     || Speed < ?NORMAL_THRESHOLD] ++
+    [{fast, ?CALL(accelerate, [accelerator(Speed, fast)])}
+     || Speed < ?FAST_THRESHOLD] ++
+    [{too_fast, ?CALL(accelerate, [accelerator(Speed, too_fast)])}
+     || Speed < ?MAX_SPEED].
+
+brake_commands(Speed) ->
+  [{stopped, ?CALL(brake, [braker(Speed, stopped)])}] ++
+    [{slow, ?CALL(brake, [braker(Speed, slow)])} || Speed > 1] ++
+    [{normal, ?CALL(brake, [braker(Speed, normal)])}
+     || Speed > ?SLOW_THRESHOLD + 1] ++
+    [{fast, ?CALL(brake, [braker(Speed, fast)])}
+     || Speed > ?NORMAL_THRESHOLD + 1] ++
+    [{too_fast, ?CALL(brake, [braker(Speed, too_fast)])}
+     || Speed > ?FAST_THRESHOLD + 1].
+
+
+%% -----------------------------------------------------------------------------
+%% Properties
+%% -----------------------------------------------------------------------------
+
+
+%% This should "never" fail:<br/>
+%% `proper:quickcheck(car_fsm:prop_normal_distance(), 1000).'
+prop_distance() ->
+  ?FORALL(Cmds, proper_fsm:commands(?MODULE),
+          ?TRAPEXIT(
+             begin
+               start_link(),
+               {_H, {_, S}, R} = proper_fsm:run_commands(?MODULE, Cmds),
+               stop(),
+               #state{distance = Distance, burnt = Burnt} = S,
+               Consumption = case Distance > 0 of
```

Comment:
The
```erlang
case Distance > 0 of
   ...
```
part gets repeated 3 times in these properties.
Perhaps it should be abstracted to a function (`calculate_consumption`?)
Should that function also include the previous line and take `S` as argument returning `{Distance, Consumption}` ?